Zotero/Scenes/Detail/DetailCoordinator.swift

```diff
@@ -974,8 +981,18 @@ extension DetailCoordinator: DetailNoteEditorCoordinatorDelegate {
         case .edit(let key), .readOnly(let key):
             DDLogInfo("DetailCoordinator: show note \(key)")
         }
-        let controller = createNoteController(library: library, kind: kind, text: text, tags: tags, parentTitleData: parentTitleData, title: title, saveCallback: saveCallback)
+        let (controller, viewModel) = createNoteController(library: library, kind: kind, text: text, tags: tags, parentTitleData: parentTitleData, title: title)
         navigationController.present(controller, animated: true)
+
+        if let saveCallback {
+            viewModel.stateObservable
+                .observe(on: MainScheduler.instance)
+                .subscribe(onNext: { [weak self] state in
+                    guard let self, state.changes.contains(.saved), case .edit(let key) = state.kind else { return }
```

Comment:
`[weak self]` is not used at all in this closure.
In any case, I think this callback will only be used by multiple open items, to create an open item for a newly created note. So I may further simplify this and remove the callback completely, if possible. E.g. doing so in the action handler instead.

```suggestion
                .subscribe(onNext: { state in
                    guard state.changes.contains(.saved), case .edit(let key) = state.kind else { return }
```

Reply:
This callback is actually used by item detail to know when a note was created/updated, so we'll probably have to keep it. Anyway I removed `[weak self]`.

Reply:
😅 I thought this one was also replaced by a direct observer, my bad.